WcaOnRails/app/models/user.rb

```diff
@@ -685,9 +719,27 @@ def editable_fields_of_user(user)
     if user.wca_id.blank? && organizer_for?(user)
       fields << :name
     end
+    if user.any_kind_of_staff?
+      fields += %i(receive_delegate_reports)
+    end
     fields
   end
 
+  def self.clear_receive_delegate_reports_if_not_staff
+    User.where(receive_delegate_reports: true).reject(&:any_kind_of_staff?).map { |u| u.update(receive_delegate_reports: false) }
+  end
+
+  # This method is only called in sync_mailing_lists_job.rb, right after clear_receive_delegate_reports_if_not_staff.
+  # If used without calling clear_receive_delegate_reports_if_not_staff it might return non-current Staff members.
+  def self.delegate_reports_receivers
+    candidate_delegates = User.candidate_delegates
+    senior_delegates = User.senior_delegates
+    wqac_members = Team.wqac.current_members.includes(:user).map(&:user)
+    wrc_members = Team.wrc.current_members.includes(:user).map(&:user)
+    other_staff = User.where(receive_delegate_reports: true)
+    (candidate_delegates + senior_delegates + wqac_members + wrc_members + other_staff).uniq
```

Comment:
How do you feel about changing this to recursively use our other already defined groups? I feel like that would reduce the number of api calls we have to make to google, and would also keeps things cleaner if someone goes looking at group membership in the future. I'm thinking something like this:

```ruby
%w(
  candidates@worldcubeassociation.org
  seniors@worldcubeassociation.org
  quality@worldcubeassociation.org
  regulations@worldcubeassociation.org
) + other_staff
```

Reply:
That sounds good! I think I'm not keeping `candidates@` though, as the only purpose of that group was to be part of `reports@`.